dolphin/db/sqlalchemy/api.py

```diff
@@ -241,11 +246,8 @@ def storage_update(context, storage_id, values):
     """Update a storage device with the values dictionary."""
     session = get_session()
     with session.begin():
-        query = _storage_get_query(context, session)
-        result = query.filter_by(id=storage_id).update(values)
-        if not result:
-            raise exception.StorageNotFound(id=storage_id)
-    return result
+        result = _access_info_get(context, storage_id, session).update(values)
```

Comment:
we may need to add exception, incase access_info doesn't found.
Ex: for pool

        if not result:
            raise exception.PoolNotFound(id=pool_id)

Reply:
check is already there in _access_info_get to raise exception when access_info not found